### PR TITLE
Indicator relevant period types

### PR DESCRIPTION
**Pull request by AlexCLeduc**
1. update phac-helpers to get new comma-separate alternative to m2m field, great for small choices like these that don't deserve their own lookups
2. add field to indicator
3. add to form
    - also added help text for this field and the relevant-dimensions, we can add a bunch more for HSO's sake
4. added period-filtering based on this var to the indicator-home view
5. There some unrelated autoformating changes, don't know how they got through before 🤷 

Here's what help-text looks like:
Screenshot 2024-02-22 at 14 47 25

### CODE REVIEW

File: server/server/fields.py [hunk 1]
```diff
@@ -12,6 +12,8 @@
 from django.utils.html import mark_safe
 from django.utils.translation import gettext_lazy as _
 
+from phac_aspc.django.fields import CommaSeparatedCharField
+
 migration_ignored_attrs = [
     "help_text",
```
Comment (AlexCLeduc) [inline server/server/fields.py:18]: Updated phac-helpers include all of the field subclasses, I might open up another PR later to just use those and we can get rid of this module (though we'd have to keep the CKEditor one)

File: server/cpho/models/indicators.py [hunk 2]
```diff
@@ -264,6 +276,38 @@ def __str__(self):
...
+            ):
+                relevant.append(period)
+            elif (
+                period.year_type == Period.CALENDAR_YEAR_TYPE
+                and period.quarter is None
+                and "calendar_years" in self.relevant_period_types
+            ):
+                relevant.append(period)
+
+        return relevant
```
Comment (AlexCLeduc) [inline server/cpho/models/indicators.py:309]: Note that if the options are left empty, it doesn't do any filtering at all, so this shouldn't really break anything

File: server/cpho/views/indicators.py [hunk 3]
```diff
@@ -44,6 +44,9 @@ class IndicatorForm(ModelForm):
     class Meta:
         model = Indicator
         fields = "__all__"
+        widgets = {
+            "relevant_period_types": forms.CheckboxSelectMultiple,
+        }
 
     name = forms.CharField(
         required=False, widget=forms.TextInput(attrs={"class": "form-control"})
```
[the remark this answers is not in the document]
Comment (msarar) [reply server/cpho/views/indicators.py:52]: I agree this is simpler. However adding it explicitly might be nice as we have control over if it's required or not (not really a big deal either way hehe)

[the remark this answers is not in the document]
Comment (AlexCLeduc) [reply server/cpho/views/indicators.py:52]: Agreed, I had an issue with that until I set blank=True in the model

**Review by msarar — APPROVED** [review]
Looks Good!
Thanks Alex !
:)

**Pull request merged**